src/neo/Network/P2P/RemoteNode.ProtocolHandler.cs

```diff
@@ -107,6 +108,10 @@ private void OnMessage(Message msg)
                     OnPongMessageReceived((PingPayload)msg.Payload);
                     break;
                 case MessageCommand.Transaction:
+                    while (Blockchain.Singleton.isPersisting)
```

Comment:
This could be contraproducente, if you wait for the incoming tx, you will reduce the TPS

Reply:
By experiments such modification does enhance TPS though..
I think things might be like this: The amount of verified transactions during persisting is very limited, as persisting also takes a lot of CPU resource. And on the other hand persisting is so long that overall TPS is lowerd.

Reply:
Hi Shargon, I changed the strategy to making remotenode stashing transactions while blockchain persisting block, please have a check:)